i18n/i18n.go

```diff
@@ -54,14 +55,15 @@ func (i18n *I18n) AddTransaltion(translation *Translation) {
 	i18n.Translations[translation.Locale][translation.Key] = translation
 }
 
-func (i18n *I18n) SaveTransaltion(translation *Translation) {
+func (i18n *I18n) SaveTranslation(translation *Translation) (err error) {
 	if i18n.Translations[translation.Locale] == nil {
 		i18n.Translations[translation.Locale] = map[string]*Translation{}
 	}
 	i18n.Translations[translation.Locale][translation.Key] = translation
```

Comment:
If we failed to save translation, then should not append the transactions to the cache map?